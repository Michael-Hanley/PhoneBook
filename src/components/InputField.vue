<template>
  <div>
    <div class="custom-input-label">
      <label :for="inputId" v-if="label">{{ label }}</label>
    </div>
    <input
      ref="inputRef"
      :id="inputId"
      :autocomplete="autocomplete"
      class="custom-input-field"
      v-model="localValue"
      @input="$emit('input', $event.target.value)"
      :placeholder="placeholder"
    />
  </div>
</template>

<script>
export default {
  name: "InputField",
  props: ["value", "label", "inputId", "placeholder", 'autocomplete'],
  data() {
    return {
      localValue: this.value,
    };
  },
  watch: {
    localValue: function () {
      this.localValue = this.value;
    },
  },
};
</script>

<style scoped>
.custom-input-field {
  box-sizing: border-box;
  width: 100%;
  padding: 5px;
  margin: 5px;
  border: none;
  font-size: inherit;
  outline: none;
  border-bottom: 1px solid gray;
}
.custom-input-field:focus {
  margin-bottom: 4px;
  border-bottom: 2px solid black;
}
.custom-input-label {
  margin-right: 10px;
  text-align: left;
}
</style>



