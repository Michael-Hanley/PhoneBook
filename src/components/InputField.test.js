import { shallowMount } from '@vue/test-utils';
import InputField from './InputField.vue';

let wrapper;
describe('InputField', () => {
  beforeEach(() => {
    wrapper = shallowMount(InputField, {
      slots: {
        default: 'test'
      },
      propsData: {
        value: 'test-value',
        label: 'test-label'
      }
    })
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});