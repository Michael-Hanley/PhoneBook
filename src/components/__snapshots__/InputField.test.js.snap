// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correctly 1`] = `
<div>
  <div class="custom-input-label"><label>test-label</label></div> <input class="custom-input-field">
</div>
`;
